app/scripts/controllers/provider-approval.js

```diff
@@ -41,7 +41,7 @@ class ProviderApprovalController extends SafeEventEmitter {
         return
       }
       // register the provider request
-      const metadata = await getSiteMetadata(origin)
+      const metadata = /localhost|[.:/]/.test(origin) ? await getSiteMetadata(origin) : { name: 'An extenstion' }
```

Comment:
Interesting method of detecting extension URLs. I think it should work for now? 🤔 
It doesn't seem ideal though. I'd prefer that we check the actual scheme of the origin, but unfortunately that's stripped out earlier on.

I'm tempted to try updating `background.js` to _include_ the scheme in the origin it passes through, then you could just extract the scheme here and use that instead. It'll be a bit delicate though, as everything else is expecting the origin to _not_ have a scheme...

Reply:
Ya, I noticed this was a bit delicate, so this is a simpler way that doesn't really break anything. Although, if the alphanumeric extension id happens to contain 'localhost', well, edge case...

Reply:
I'm tempted to accept the regex for now and improve this using the `scheme` later on, but... I'm concerned there will be more edge cases than just `localhost`. It's not uncommon to use custom domains locally for testing for example (perhaps by modifying the `hosts` file).

Instead I think we can sidestep even attempting to detect extensions by using the origin as the fallback if a title isn't present (as mentioned in my other comment). This lets us avoid this until we can do it well.

Reply:
> It's not uncommon to use custom domains locally for testing
So long as the the custom domain in the hosts file has a tld, or if a port is used, it will be picked up as a domain

> Instead I think we can sidestep even attempting to detect extensions by using the origin as the fallback if a title isn't present
Keep in mind, this isn't detecting an extension, its detecting domains with a very large net. Also, I am not sure what you mean by "title". Do you mean the `name` in the object returned by `getSiteMetadata`? If so, the issue is that `getSiteMetadata` throws silently for extensions, and never returns `null` for `name`.

To be honest, I didn't really understand how `getSiteMetadata` gets what it gets, as the code got pretty consulted, and wasn't that functions. At least IMO.

Reply:
>So long as the the custom domain in the hosts file has a tld, or if a port is used, it will be picked up as a domain

You're right about the `tld`. I don't think the port is included though - the `origin` parameter here is really just the `hostname`, which afaik doesn't include the port.

>I am not sure what you mean by "title". Do you mean the name in the object returned by getSiteMetadata?
Sorry yes, that's what I meant.

>the issue is that getSiteMetadata throws silently for extensions, and never returns null for name.

Ah I see 🤔. I had not realized - that does change things. I'll take a quick look to see if I can find out why this happens.

Reply:
Maybe we/I modify `getSiteMetadata` to do the regex test before calling its downstream functions, and immediately return a null name and icon, or a null object altogether, so that we can ignore why downstream calls might crash. The we can implement the UI logic as you suggested.

Reply:
I've made an attempt here: https://github.com/MetaMask/metamask-extension/commit/1b328016fc0bff7ef1a562c116c655caf4fc0277

I'm not _entirely_ sure that the `extensionId` is a reliable indicator of whether the sender is an extension or not; the docs imply it'll bet set for extensions but I haven't tested it yet. We could check the scheme instead (for `chrome-extension` or `moz-extension`, and that should work reliably as well. It does avoid calling `getSiteMetadata` just as your solution did, so that;s good.

The tests still aren't working; I need to investigate a bit to find out why. I haven't updated the UI yet either, so it'll still likely break because it expects a name in the metadata.